homeassistant/components/broadlink/sensor.py

```diff
@@ -89,6 +136,59 @@ def __init__(self, device, monitored_condition):
         self._attr_unique_id = f"{device.unique_id}-{monitored_condition}"
         self._attr_unit_of_measurement = SENSOR_TYPES[monitored_condition][1]
 
+    async def async_added_to_hass(self):
+        """Call when the entity is added to hass."""
+        await super().async_added_to_hass()
+
+        if not (
+            self.device_class == DEVICE_CLASS_ENERGY
+            and self.state_class == STATE_CLASS_MEASUREMENT
+        ):
+            return
+
+        # Summing sensors require a last_reset attribute if used together
+        # with state_class. Here we update the last_reset with the data
+        # restored from the registry, when applicable.
+
+        state = await self.async_get_last_state()
+        if state is None or self._did_reset(state.state, self._attr_state):
+            self._attr_last_reset = dt.utcnow()
+        else:
+            last_reset = state.attributes[ATTR_LAST_RESET]
+            self._attr_last_reset = dt.as_utc(datetime.fromisoformat(last_reset))
+
     def _update_state(self, data):
         """Update the state of the entity."""
-        self._attr_state = data[self._monitored_condition]
+        new_state = data[self._monitored_condition]
+
+        if (
+            self.device_class == DEVICE_CLASS_ENERGY
+            and self.state_class == STATE_CLASS_MEASUREMENT
+            and self._did_reset(self._attr_state, new_state)
+        ):
+            self._attr_last_reset = dt.utcnow()
+
+        self._attr_state = new_state
+
+    @staticmethod
+    def _did_reset(last_state, new_state):
+        """Return True if the accumulator has reset.
+
+        Compare the last state with the new one. If the value has reset
+        to 0 or has significantly reduced we say it has reset.
+        """
+        if last_state in (STATE_UNKNOWN, STATE_UNAVAILABLE):
+            # First measurement we ever got for this sensor. Always a reset.
+
+            # If the last state is unavailable we cannot determine the last
+            # reset, so we consider it a reset in order to start over.
+            return True
+
+        if new_state in (STATE_UNKNOWN, STATE_UNAVAILABLE):
```

Comment:
This conditional is never reached in the current implementation, but I thought it would be nice to have. We cannot test it though. Keep it or remove it?